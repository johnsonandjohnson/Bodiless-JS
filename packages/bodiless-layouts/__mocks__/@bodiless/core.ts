const editContext = {
  activate: jest.fn(),
  isEdit: true,
};

export const useEditContext = jest.fn(() => editContext);

const activateOnEffect = {
  setId: jest.fn(),
};
export const useActivateOnEffect = () => activateOnEffect;

export const useNode = jest.fn();

const contextMenuFormInner = jest.fn();
export const contextMenuForm = jest.fn(() => contextMenuFormInner);
