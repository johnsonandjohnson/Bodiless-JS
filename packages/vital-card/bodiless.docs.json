{
  "VitalDesignSystem": {
    "Components": {
      "VitalCard": {
        "README.md": "./doc/README.md",
        "assets": {
          "HeroSwapHeroCard.jpg": "./doc/assets/HeroSwapHeroCard.jpg"
        }
      }
    }
  }
}
