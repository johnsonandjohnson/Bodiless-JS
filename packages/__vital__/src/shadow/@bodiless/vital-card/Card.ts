/**
 * Copyright Â© 2022 Johnson & Johnson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { vitalCardBase } from '@bodiless/vital-card';
import { asFluidToken } from '@bodiless/vital-elements';
import {
  Div,
  addProps,
  startWith,
  flowHoc
} from '@bodiless/fclasses';

const Default = asFluidToken({
  ...vitalCardBase.Default,
  Components: {
    ...vitalCardBase.Default.Components,
    Wrapper: addProps({ 'data-shadowed-by': '__vitalstarter_:Card' }),
  },
});

// TODO not rendering
const Hero = asFluidToken({
  ...vitalCardBase.Hero,
  Components: {
    ...vitalCardBase.Hero.Components,
    Wrapper: flowHoc(
      startWith(Div),
      addProps({ 'data-shadowed-by': '__vitalstarter_:HeroCard' }),
    ),
  },
});

export default {
  ...vitalCardBase,
  Default,
  Hero,
};
