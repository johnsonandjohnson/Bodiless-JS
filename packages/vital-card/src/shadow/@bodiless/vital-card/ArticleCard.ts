import { shadow } from '@bodiless/vital-elements';
import { vitalArticleCard } from '../../../components/ArticleCard';

export default shadow(vitalArticleCard, 'Vital:ArticleCard');
