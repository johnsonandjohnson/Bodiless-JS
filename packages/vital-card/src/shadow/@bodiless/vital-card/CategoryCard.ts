import { shadow } from '@bodiless/vital-elements';
import { vitalCategoryCard } from '../../../components/CategoryCard';

export default shadow(vitalCategoryCard, 'vitalCard:CategoryCard');
