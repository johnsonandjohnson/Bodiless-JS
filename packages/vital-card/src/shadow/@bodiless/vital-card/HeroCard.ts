import { shadow } from '@bodiless/vital-elements';
import { vitalHeroCard } from '../../../components/HeroCard';

export default shadow(vitalHeroCard, 'Vital:HeroCard');
