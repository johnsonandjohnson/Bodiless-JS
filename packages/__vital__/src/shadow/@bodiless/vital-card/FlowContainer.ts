/**
 * Copyright Â© 2022 Johnson & Johnson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { as } from '@bodiless/fclasses';
import { vitalCardFlowContainerBase } from '@bodiless/vital-card';

// Demostrating only showing Basic & Hero Card Variations in Component Picker
const WithCardVariations = as(
  vitalCardFlowContainerBase.WithBasicVariations,
  vitalCardFlowContainerBase.WithHeroVariations,
);

// const WithCardVariations = vitalCardFlowContainerBase.WithCardVariations;

export default {
  ...vitalCardFlowContainerBase,
  WithCardVariations,
};
