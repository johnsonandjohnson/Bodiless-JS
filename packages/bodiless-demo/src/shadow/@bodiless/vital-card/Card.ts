import { asCardToken, vitalCardBase } from '@bodiless/vital-card';

const Hero = asCardToken(vitalCardBase.Hero, {
  // ...vitalCardBase.WithPrimaryButton,
});

export default {
  ...vitalCardBase,
  Hero,
};
