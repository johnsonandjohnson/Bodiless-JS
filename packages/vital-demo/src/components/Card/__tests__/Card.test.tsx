import React from 'react';
import { testTokens } from '@bodiless/vital-elements';
import { CardClean } from '@bodiless/vital-card';
import { demoCard } from '..';

// eslint-disable-next-line jest/valid-describe
describe('Card Tokens', testTokens(CardClean, demoCard));
