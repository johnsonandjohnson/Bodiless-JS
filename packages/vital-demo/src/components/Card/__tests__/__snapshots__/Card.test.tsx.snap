// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Tokens Matches snapshot for Category 1`] = `
<a
  className="w-full flex flex-col"
  href="#"
>
  <div>
    <div
      data-no-hydrate={true}
      id="f64e1b3016f7c8d0d277ce020d98ee0d"
      style={
        Object {
          "display": "contents",
        }
      }
    >
      <img
        alt=""
        src="test-file-stub"
        title=""
      />
    </div>
  </div>
  <div>
    <h3
      className="text-vital-primary-header-copy mb-5 font-medium text-m-xl lg:text-xl"
    >
      <span
        data-no-hydrate={true}
        id="6e5e36380662b0a36aed30babe228408"
        style={
          Object {
            "display": "contents",
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Card Title",
            }
          }
        />
      </span>
    </h3>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for Topic 1`] = `
<a
  className="w-full flex flex-col"
  href="#"
>
  <div>
    <div
      data-no-hydrate={true}
      id="f64e1b3016f7c8d0d277ce020d98ee0d"
      style={
        Object {
          "display": "contents",
        }
      }
    >
      <img
        alt=""
        src="test-file-stub"
        title=""
      />
    </div>
  </div>
  <div>
    <div
      className="text-vital-secondary-eyebrow font-extrabold uppercase mb-3 text-m-xs lg:text-xs"
    >
      <span
        data-no-hydrate={true}
        id="a315e6de1b5f67d75da9c7058677c509"
        style={
          Object {
            "display": "contents",
          }
        }
      >
        <span
          className="my-4"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Card Eyebrow",
            }
          }
        />
      </span>
    </div>
    <h3
      className="text-vital-primary-header-copy mb-5 font-medium text-m-xl lg:text-xl"
    >
      <span
        data-no-hydrate={true}
        id="6e5e36380662b0a36aed30babe228408"
        style={
          Object {
            "display": "contents",
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Card Title",
            }
          }
        />
      </span>
    </h3>
    <div>
      <a>
        <span>
          <span
            data-no-hydrate={true}
            id="adfc8145bf1b71ec6355d197fef5660d"
            style={
              Object {
                "display": "contents",
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Call To Action Link",
                }
              }
            />
          </span>
        </span>
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithFlexGrowImage 1`] = `
<a>
  <div
    className="flex-grow"
  >
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithFlowContainerPreview 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithHorizontalContentAtTop 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div
      className="flex-grow"
    />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithHorizontalContentCentered 1`] = `
<a>
  <div>
    <img />
  </div>
  <div
    className="md:content-center md:justify-center md:grow"
  >
    <div />
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithHorizontalLeftOrientation 1`] = `
<a
  className="flex-col md:flex-row w-full flex"
>
  <div
    className="py-0 md:py-0 md:w-1/2 flex flex-col"
  >
    <img
      className="w-full"
    />
  </div>
  <div
    className="px-4 md:w-1/2 flex flex-col"
  >
    <div />
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithHorizontalOrientationBase 1`] = `
<a>
  <div
    className="py-0 md:py-0 md:w-1/2 flex flex-col"
  >
    <img
      className="w-full"
    />
  </div>
  <div
    className="px-4 md:w-1/2 flex flex-col"
  >
    <div />
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithHorizontalRightOrientation 1`] = `
<a
  className="flex-col md:flex-row-reverse w-full flex"
>
  <div
    className="py-0 md:py-0 md:w-1/2 flex flex-col"
  >
    <img
      className="w-full"
    />
  </div>
  <div
    className="px-4 md:w-1/2 flex flex-col"
  >
    <div />
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithNoDescription 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithNoEyebrow 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithNoTitle 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithPrimaryButton 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div>
      <a
        className="pl-12 pr-6 py-3.5 group flex-row-reverse text-white bg-vital-primary-interactive hover:opacity-80 leading-tight uppercase focus:outline-none focus:ring-0 rounded transition duration-150 ease-in-out px-6 flex justify-center"
        href="#"
        role="button"
      >
        <span
          className="w-6 h-2 vital-arrow group-hover:text-current text-transparent pr-1 inline-block"
        />
        <span
          data-no-hydrate={true}
          id="123b58b4a9e4407d42b2606d912b6deb"
          style={
            Object {
              "display": "contents",
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "Link",
              }
            }
          />
        </span>
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithPrimaryTextLink 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div>
      <a
        className="hover:vital-arrow hover:pr-1 hover:w-6 hover:h-2 text-vital-primary-interactive hover:opacity-70 active:text-vital-primary-interactive-active underline font-bold"
        href="#"
      >
        <span
          className="text-vital-primary-interactive active:text-vital-primary-interactive-active"
        />
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithSecondaryButton 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div>
      <a
        className="pl-12 pr-6 py-3.5 group flex-row-reverse border-vital-primary-interactive text-vital-primary-interactive hover:opacity-70 active:text-vital-primary-interactive-active border-2 hover:opacity-80 leading-tight uppercase focus:outline-none focus:ring-0 rounded transition duration-150 ease-in-out px-6 flex justify-center"
        href="#"
        role="button"
      >
        <span
          className="w-6 h-2 vital-arrow group-hover:text-current text-transparent pr-1 inline-block"
        />
        <span
          data-no-hydrate={true}
          id="123b58b4a9e4407d42b2606d912b6deb"
          style={
            Object {
              "display": "contents",
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "Link",
              }
            }
          />
        </span>
      </a>
    </div>
  </div>
</a>
`;

exports[`Card Tokens Matches snapshot for WithVerticalOrientation 1`] = `
<a
  className="w-full flex h-full flex-col"
>
  <div
    className="py-4 md:py-8"
  >
    <img
      className="w-full"
    />
  </div>
  <div
    className="py-4"
  >
    <div />
    <h3 />
    <div
      className="flex-grow"
    />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;
